src/ipfs.rs

```diff
@@ -98,7 +98,7 @@ impl ContentAddressedStorage for Ipfs {
     }
     async fn list(&self) -> Result<Vec<Cid>, Self::Error> {
         // return a list of all CIDs which are aliased/pinned
-        self.list_pins(Some(PinMode::Recursive))
+        self.list_pins(Some(PinMode::Direct))
```

Comment:
Are you sure about this? We should only use recursive pins, and `PinMode::Recursive` should only return the "root" block of the pin, all children should have `PinMode::Indirect`

Reply:
Not certain, currently the blocks pinned in this api are pinned non-recursively, so they aren't returned with Recursive. I wouldn't mind pinning recursively above and returning recursive pins here